src/social-web/components/site-hub/index.tsx

```diff
@@ -34,16 +38,45 @@ function SiteHub() {
 		};
 	}, [] );
 
-	const { open: openCommandCenter } = useDispatch( commandsStore );
+	// Track hash changes to trigger re-renders
+	const [ currentHash, setCurrentHash ] = useState( window.location.hash );
+
+	useEffect( () => {
+		const handleHashChange = () => {
+			setCurrentHash( window.location.hash );
+		};
+
+		window.addEventListener( 'hashchange', handleHashChange );
+		return () => {
+			window.removeEventListener( 'hashchange', handleHashChange );
+		};
+	}, [] );
```

Comment:
I think we found the limits of our simple router setup.

Asking to switch to Tanstack before moving forward with these styles feels like a lot, but this introduces quite a bit of tech debt otherwise. What do you think?